pkg/distribution/internal/store/store_test.go

```diff
@@ -382,6 +382,88 @@ func TestStoreAPI(t *testing.T) {
 	})
 }
 
+func TestWriteRollsBackOnTagFailure(t *testing.T) {
```

Comment:
Thanks for adding a test of tag failure, could you also add one for config failure and layer failure? 🙏 
Ideally we should also add test of failure with an already existing blob but we can do it in a follow up PR